discovery-provider/src/tasks/index.py

```diff
@@ -351,355 +376,284 @@ def update_ursm_address(self):
             )
 
 
-def save_and_get_skip_tx_hash(session, redis):
+def get_tx_hash_to_skip(session, redis):
     """Fetch if there is a tx_hash to be skipped because of continuous errors"""
     indexing_error = get_indexing_error(redis)
     if (
         isinstance(indexing_error, dict)
         and "has_consensus" in indexing_error
         and indexing_error["has_consensus"]
     ):
-        try:
-            add_network_level_skipped_transaction(
-                session,
-                indexing_error["blocknumber"],
-                indexing_error["blockhash"],
-                indexing_error["txhash"],
-            )
-        except Exception:
-            return None
         return indexing_error["txhash"]
-    return None
-
+    else:
+        return None
 
-def index_blocks(self, db, blocks_list):
-    web3 = update_task.web3
-    redis = update_task.redis
 
-    num_blocks = len(blocks_list)
-    block_order_range = range(len(blocks_list) - 1, -1, -1)
-    latest_block_timestamp = None
-    for i in block_order_range:
-        update_ursm_address(self)
-        block = blocks_list[i]
-        block_index = num_blocks - i
-        block_number = block.number
-        block_hash = block.hash
-        block_timestamp = block.timestamp
-        latest_block_timestamp = block_timestamp
-        logger.info(
-            f"index.py | index_blocks | {self.request.id} | block {block.number} - {block_index}/{num_blocks}"
+def save_skipped_tx(session, redis):
+    indexing_error = get_indexing_error(redis)
+    try:
+        add_network_level_skipped_transaction(
+            session,
+            indexing_error["blocknumber"],
+            indexing_error["blockhash"],
+            indexing_error["txhash"],
+        )
+    except Exception:
+        logger.warning(
+            f"index.py | save_skipped_tx: Failed to add_network_level_skipped_transaction for {indexing_error}"
         )
-        challenge_bus: ChallengeEventBus = update_task.challenge_event_bus
-        with db.scoped_session() as session, challenge_bus.use_scoped_dispatch_queue():
-            current_block_query = session.query(Block).filter_by(is_current=True)
 
-            # Without this check we may end up duplicating an insert operation
-            block_model = Block(
-                blockhash=web3.toHex(block.hash),
-                parenthash=web3.toHex(block.parentHash),
-                number=block.number,
-                is_current=True,
+
+# Append tx to list of txs of its contract type.
```

Comment:
Nit, we should use multi line string docstrings for bigger comments

```
def group_***:
   """
   Append tx to list of ...
   """
```


https://google.github.io/styleguide/pyguide.html#381-docstrings